modules/photons_control/clean.py

```diff
@@ -0,0 +1,189 @@
+
+    async def gen(ref, sender, **kwargs):
+        r = ref if reference is None else reference
+
+        plans = sender.make_plans("capability")
+        async for serial, _, info in sender.gatherer.gather(plans, r, **kwargs):
+            if info["cap"].has_hev:
+                yield LightMessages.SetHevCycleConfiguration(
+                    indication=bool(indication),
+                    duration_s=duration,
+                    target=serial,
+                    ack_required=True,
+                    res_required=False,
+                )
+
+    return FromGenerator(gen)
+
+
+@task
+class set_clean_default_config(task.Task):
+    """
+    Set the default clean cycle configuration for a device
+
+    ``target:clean_config match:cap=hev -- '{"indication": true, "duration": 7200}'``
+
+    It takes an "indication" field that is a boolen which indicates whether to
+    run a short flashing indication at the end of the HEV cycle. It also takes
+    a ``duration`` field that is the seconds of the duration.
+    """
+
+    target = task.requires_target()
+    reference = task.provides_reference(special=True)
+
+    async def execute_task(self, **kwargs):
+        options = self.photons_app.extra_as_json
+        options = sb.set_options(
+            indication=sb.required(sb.boolean()),
+            duration=sb.required(sb.integer_spec()),
+        ).normalise(Meta.empty(), options)
+
+        await self.target.send(SetCleanConfig(**options), self.reference, **kwargs)
+
+
+@task
+class get_clean_default_config(task.Task):
+    """
+    Get the default HEV cleaning cycle configuration for the target bulb
+    """
+
+    target = task.requires_target()
+    reference = task.provides_reference(special=True)
+
+    async def execute_task(self, **kwargs):
+        async def gen(reference, sender, **kwargs):
+            ps = sender.make_plans("capability")
+            async for serial, _, info in sender.gatherer.gather(ps, reference, **kwargs):
+                if info["cap"].has_hev:
+                    yield LightMessages.GetHevCycleConfiguration(target=serial)
+
+        async for pkt in self.target.send(FromGenerator(gen), self.reference):
+            print(pkt.serial)
+            indication = "enabled" if pkt.indication else "disabled"
+            print(f"   Cycle time: {pkt.duration_s}")
+            print(f"   Indication: {indication} (raw: {pkt.indication})")
+            print("")
+
+
+def StartStopCleanCycle(enable=None, duration=None, reference=None, **kwargs):
+    """
+    Returns a valid message that will either start or stop the cleaning (HEV)
+    cycle of devices used against it.
+
+    When a cycle is started, if the duration=0 or is not provided, the default
+    duration will be used.
+
+    For example:
+
+    .. code-block:: python
+
+        await target.send(CleanCycle(), ["d073d5000001", "d073d5000001"])
+    """
+
+    async def gen(ref, sender, **kwargs):
+        r = ref if reference is None else reference
```

Comment:
you should still use the data normalisation here to have real assurance of what the values are